File: apps/mybookkeeper/frontend/src/shared/types/lease/lease-template-summary.ts
/**
 * Minimal lease-template payload for list views.
 *
 * Mirrors `schemas/leases/lease_template_summary.py`.
 */
export interface LeaseTemplateSummary {
  id: string;
  user_id: string;
  organization_id: string;
  name: string;
  description: string | null;
  version: number;
  file_count: number;
  placeholder_count: number;
  created_at: string;
  updated_at: string;
}

File: apps/mybookkeeper/backend/app/schemas/leases/lease_template_summary.py
"""Minimal lease-template payload for list views."""
from __future__ import annotations

import datetime as _dt
import uuid

from pydantic import BaseModel, ConfigDict


class LeaseTemplateSummary(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    organization_id: uuid.UUID
    name: str
    description: str | None = None
    version: int
    file_count: int
    placeholder_count: int
    created_at: _dt.datetime
    updated_at: _dt.datetime

    model_config = ConfigDict(from_attributes=True)
